main.py

```diff
@@ -45,77 +49,100 @@ def get_eurojackpot_results():
+        if number in guesses_secondary:
             secondary_hits += 1
 
-    result = f"{primary_hits}+{secondary_hits} oikein"
+    total_hits = f"{primary_hits}+{secondary_hits} oikein"
 
     money_won = 0
-
-    for tier in prize_tiers:
-        if tier["name"] == result:
-            money_won = int(tier["shareAmount"])
+    for prize_tier in eurojackpot.prize_tiers:
+        if prize_tier.name == total_hits:
+            money_won = prize_tier.share_amount
+            break
 
     investment_value_old = get_investment_value()
-    investment_value = str(investment_value_old - 200 + money_won)
+    investment_value_new = investment_value_old - 200 + money_won
+
+    set_investment_value(investment_value_new)
+
+    return primary_hits, secondary_hits, money_won, investment_value_new
+
 
-    set_investment_value(investment_value)
+def generate_discord_msg(guesses_primary: List[Union[int, str]], guesses_secondary: List[Union[int, str]]) -> str:
+    guesses_primary = [str(i) for i in guesses_primary]
+    guesses_secondary = [str(i) for i in guesses_secondary]
 
-    return f"<@&{discord_group_id}> {result}, voittoa {int(money_won)/100:.2f}€, sijoituksen tuotto ||{int(investment_value)/100:.2f}€||"
+    ejackpot_messages = []
+    results = get_eurojackpot_results()
+    if not results:
+        return "Tuloksia ei saatu Veikkaukselta :("
+
+    for result in results:
+        info = fetch_winnings(result, guesses_primary, guesses_secondary)
+
+        primary_hits = info[0]
+        secondary_hits = info[1]
+        money_won = info[2]
+        investment_value = info[3]
+
+        ejackpot_week = datetime.datetime.fromtimestamp(result.close_time/1000).isocalendar().week
+        game = result.brand_name.split("-")[1]
+        weekday = result.brand_name.split("-")[0]
+        msg = f"{game} viikko {ejackpot_week}/{weekday}, " \
+              f"{primary_hits}+{secondary_hits} oikein, " \
+              f"voittoa {int(money_won)/100:.2f}€, " \
+              f"sijoituksen tuotto ||{investment_value/100:.2f}€||"
+        ejackpot_messages.append(msg)
+
+    return "\n".join(ejackpot_messages)
 
 
 def lambda_handler(event=None, context=None):
     asyncio.run(client.start(discord_key))
 
 
-if __name__ == "__main__":
-    if Path("env.json").is_file():
-        env_variables = json.load(open("env.json"))
+def fetch_env_variables() -> None:
+    with open("env.json", "r") as env_file:
+        env_variables = json.load(env_file)
         for variable_name, variable_value in env_variables.get("Variables").items():
             os.environ[variable_name] = variable_value
 
 
-discord_key = os.environ.get("DISCORD_KEY")
-discord_channel_name = os.environ.get("DISCORD_CHANNEL_NAME")
-discord_group_id = os.environ.get("DISCORD_GROUP_ID")
-
-parameter_store_variable_name = os.environ.get("PARAMETER_STORE_VARIABLE_NAME")
+if __name__ == "__main__":
+    fetch_env_variables()
+    discord_key = os.environ.get("DISCORD_KEY")
```

Comment:
It would be tidier to have these variables under `if __name__ == "__main__":` but then the lambda function will fail. If there is a "correct" way to give arguments for `on_ready()` function I would prefer that

Reply:
I think you have resolved this issue in commit https://github.com/Glukoosi/eurojackbot/commit/5afa24a9be7aadb3d3836e0499a370b568095983?

discord.py has no way to pass parameters for `on_ready()` and by default executing any code there is not recommended, as it can fire more than once. This bot exits immediately after this function finishes so this is a little bit redundant info here and I think current way is sufficient.